tools/image_processing/yolo/json2yolosegment.py

```diff
@@ -0,0 +1,64 @@
+import argparse
+import json
+import os
+
+
+def convert_json_to_yolo(input_dir, save_dir, class_names_file):
+
+    with open(class_names_file, 'r') as f:
+        class_names = [line.strip() for line in f.readlines()]
+
+    class_to_index = {class_name: i for i, class_name in enumerate(class_names)}
+
+    for filename in os.listdir(input_dir):
+        json_filepath = os.path.join(input_dir, filename)
+
+        with open(json_filepath, 'r') as f:
+            data = json.load(f)
+
+        image_width = data.get('imageWidth')
+        image_height = data.get('imageHeight')
+        if image_width is None or image_height is None:
+            print(f"Skipping {filename}: missing image dimensions.")
+            return
+
+        annotations = data.get('shapes', [])
+
+        base, _ = os.path.splitext(filename)
+        output_file = f"{base}.txt"
+        output_filepath = os.path.join(save_dir, output_file)
+
+        with open(output_filepath, 'w') as f:
+            for annotation in annotations:
+                label = annotation.get('label')
+                class_index = class_to_index[label]
+
+                points = annotation.get('points', [])
+
+                if not points:
+                    print(f"No points found for annotation '{label}', skipping.")
+                    continue
+
+                x = [point[0] / image_width for point in points]
+                y = [point[1] / image_height for point in points]
```

Comment:
I couldn't find documentation on the format, but from the code, I suspect that `point` is in pixel coordinates, whereas `x` and `y` need to be between `0` and `1`. In that case, there probably is a small bug here:
```suggestion
                x = [point[0] / (image_width - 1) for point in points]
                y = [point[1] / (image_height - 1) for point in points]
```
This is because, for points that are in the last row/column of the image, we want to have coordinates of 1. With the original code, this wouldn't be the case due to zero-based numbering, right?

Reply:
i don't quite understand it. this is just reading from anylabeling generated json..points, image_width/height is just reading whatever output from anylabeling, so this should be already looked after

Reply:
Maybe I'm the one who misunderstands something :) lets brake it down:
**A1.** `image_width` is the width of the image, that comes from AnyLabeling?
**A2.** `point[0]` is the horizontal coordinate of a point, in pixels, that comes from AnyLabeling?
**A3.** For YOLO TXT, `x` needs to be in [0,1], where 0 is the left-most pixel and 1 is the right-most pixel of the image?

If the above is true, then, lets see what we get with the original code:
- Suppose `image_width` is 10
- Suppose `point[0]` is 0 (corresponds to a left-most pixel)
- Then we get `x = 0` which is fine ✅
- Now, suppose `point[0]` is 9 (corresponds to the right-most pixel)
- Then we get `x = 0.9` which contradicts **A3** from above. ❌

With the fixed code, and `image_width = 10` as an example, we get:
- `x = 0` if `point[0]` is 0, i.e. a left-most pixel ✅
- `x = 1` if `point[0]` is 9, i.e. a right-most pixel ✅

Reply:
sorry i still don't understand. 
A1,2,3 -> yes
how is x=0.9 contradicts A3?
sorry the code is written by one of our image anylysts, i would have to ask him.

Reply:
ok. understood. Thanks a lot. i will change the code